sdk/src/main/scala/com/horizen/companion/SidechainTransactionsCompanion.scala

```diff
@@ -2,18 +2,24 @@ package com.horizen.companion
 
 import com.horizen.transaction._
 import com.horizen.transaction.CoreTransactionsIdsEnum._
+
 import java.util.{HashMap => JHashMap}
 import java.lang.{Byte => JByte}
-
 import com.horizen.SidechainTypes
+import com.horizen.cryptolibprovider.utils.TypeOfCircuit
+import com.horizen.transaction.serializers.{KeyRotationTransactionSerializer, MC2SCAggregatedTransactionSerializer, OpenStakeTransactionSerializer, SidechainCoreTransactionSerializer}
 import com.horizen.utils.DynamicTypedSerializer
 
 
-case class SidechainTransactionsCompanion(customTransactionSerializers: JHashMap[JByte, TransactionSerializer[SidechainTypes#SCBT]])
-  extends DynamicTypedSerializer[SidechainTypes#SCBT, TransactionSerializer[SidechainTypes#SCBT]](
-    new JHashMap[JByte, TransactionSerializer[SidechainTypes#SCBT]]() {{
-      put(MC2SCAggregatedTransactionId.id(), MC2SCAggregatedTransactionSerializer.getSerializer.asInstanceOf[TransactionSerializer[SidechainTypes#SCBT]])
-      put(SidechainCoreTransactionId.id(), SidechainCoreTransactionSerializer.getSerializer.asInstanceOf[TransactionSerializer[SidechainTypes#SCBT]])
-      put(OpenStakeTransactionId.id(), OpenStakeTransactionSerializer.getSerializer.asInstanceOf[TransactionSerializer[SidechainTypes#SCBT]])
-    }},
+case class SidechainTransactionsCompanion(customTransactionSerializers: JHashMap[JByte, TransactionSerializer[SidechainTypes#SCBT]], typeOfCircuit: Int = TypeOfCircuit.NaiveThresholdSignatureCircuit.id)
```

Comment:
Also, here replace `Int` with `CircuitType`

Reply:
As we agreed previously, we use int instead of CircuitType as enum. Python tests are also written for int, according to the previous suggestion

Reply:
Done